### PR TITLE
T15411 logger enhancements

### LINKED ISSUE #15411: [BUG]: Logger not using Item time
The Logger component does not use the `Item` time (entry to be logged). This is technical debt from before and the interface needs to change when creating an item.

- Benchmarks need to be created to check whether `Item` (object) is more expensive than an array instead.
- The time needs to be corrected and the timezone needs to be introduced in the adapter
- The transactional mode of the logger has a loop with the queued items and writes each item in the loop. We need to change that to a processing cycle (for/next loop) and then one write - especially for the `Stream` adapter

### PR DESCRIPTION
**Pull request by niden**
Hello!

* Type: bug fix - enhancement
* Link to issue: Fixes #15411 

**In raising this pull request, I confirm the following:**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)
- [x] I have checked that another pull request for this purpose does not exist
- [x] I wrote some tests for this PR
- [x] I have updated the relevant CHANGELOG
- [ ] I have created a PR for the [documentation](https://github.com/phalcon/docs) about this change

Changes to the `Phalcon\Logger`:
    - Renamed `Phalcon\Logger\Item::getName` to `Phalcon\Logger\Item::getLevelName`   
    - Renamed `Phalcon\Logger\Item::getType` to `Phalcon\Logger\Item::getLevel`
    - Changed the logger timestamp to be `DateTimeImmutable` on a per item basis - more accurate timing
    - Changed `Phalcon\Logger` to accept a `DateTimeZone` in the constructor 
    - Changed placeholder `%type%` to `%level%` to align with the variables [#15411](https://github.com/phalcon/cphalcon/issues/15411)

Thanks

### CODE REVIEW

**Comment by Jeckerson** [comment]
Phalcon isn't ready for `0.13.2` Zephir atm.

**Comment by niden** [comment]
This is on hold until a new version of Zephir is released to correctly handle `DateTimeImmutable` and `DateTimeZone`.

cc @Jeckerson

**Comment by Jeckerson** [comment]
Error in pgsql tests occur due passing empty string schema, instead of null.

File: phalcon/Logger.zep [hunk 1]
```diff
@@ -80,21 +89,35 @@ class Logger implements LoggerInterface
     protected name = "";
 
     /**
-     * The excluded adapters for this log process
-     *
-     * @var AdapterInterface[]
+     * @var DateTimeZone
```
Comment (Jeckerson) [inline phalcon/Logger.zep:92]:
```suggestion
     * @var DateTimeZone|null
```

File: phalcon/Logger.zep [hunk 2]
```diff
@@ -80,21 +89,35 @@ class Logger implements LoggerInterface
...
+    protected timezone = null;
 
     /**
      * Constructor.
      *
-     * @param string name     The name of the logger
-     * @param array  adapters The collection of adapters to be used for logging (default [])
+     * @param string       name     The name of the logger
+     * @param array        adapters The collection of adapters to be used for logging (default [])
+     * @param DateTimeZone timezone The timezone
```
Comment (Jeckerson) [inline phalcon/Logger.zep:101]:
```suggestion
     * @param DateTimeZone|null timezone The timezone
```

File: phalcon/Logger/Formatter/AbstractFormatter.zep [hunk 3]
```diff
@@ -31,12 +31,12 @@ abstract class AbstractFormatter implements FormatterInterface
      * @param string $message
      * @param array $context
      */
-    public function interpolate(string message, var context = null)
+    public function interpolate(string message, var context = null) -> string
```
Comment (Jeckerson) [inline phalcon/Logger/Formatter/AbstractFormatter.zep:34]: Could we transform `var context = null` to `array context = []`

File: tests/unit/Logger/Item/GetMessageCest.php [hunk 4]
```diff
@@ -25,7 +26,8 @@ class GetMessageCest
     public function loggerItemGetMessage(UnitTester $I)
     {
         $I->wantToTest('Logger\Item - getMessage()');
-        $time = time();
+        $time = new DateTimeImmutable("now");
+        ;
```
Comment (Jeckerson) [inline tests/unit/Logger/Item/GetMessageCest.php:30]:
```suggestion
```

**Review by Jeckerson — APPROVED** [review]

**Pull request merged**